lib/inspec/cached_fetcher.rb

```diff
@@ -52,7 +52,19 @@ def fetch
         fetch
       elsif cache.exists?(cache_key) && !cache.locked?(cache_key)
         Inspec::Log.debug "Using cached dependency for #{target}"
-        [cache.prefered_entry_for(cache_key), false]
+        cache_value = cache.prefered_entry_for(cache_key)
+        if cache_value
+          [cache_value, false]
+        else
+          Inspec::Log.debug "Dependency does not exist in the cache for target #{target}"
+          cache_key_name = fetcher.cache_key
```

Comment:
I think the value for `fetcher.cache_key` and `cache_key` is going to be the same. So, any reason behind using `fetcher.cache_key`? 



```
(byebug) cache_key
"gem:inspec-mongodb-resources:"
(byebug) fetcher.cache_key
"gem:inspec-mongodb-resources:"
```

Although , the value for `cache_key` maybe different based on the `k`'s value in the method `cache_key`
https://github.com/inspec/inspec/blob/b896dfc3405006d86e32f6e1af02d5c6c18670a9/lib/inspec/cached_fetcher.rb#L29